src/test/java/de/tum/in/www1/artemis/ModelingExerciseIntegrationTest.java

```diff
@@ -59,4 +70,74 @@ public void getCompassStatistic_asInstructor_Forbidden() throws Exception {
+    }
+
+    @Test
+    @WithMockUser(username = "tutor1", roles = "TA")
+    public void testGetModelingExercise_asTA() throws Exception {
+        request.get("/api/modeling-exercises/" + classExercise.getId(), HttpStatus.OK, ModelingExercise.class);
+    }
+
+    @Test
+    @WithMockUser(username = "tutor1", roles = "TA")
+    public void testGetModelingExerciseForCourse_asTA() throws Exception {
+        request.get("/api/courses/" + classExercise.getCourse().getId() + "/modeling-exercises", HttpStatus.OK, List.class);
+    }
+
+    @Test
+    @WithMockUser(username = "tutor1", roles = "TA")
+    public void testGetModelingExerciseStatistics_asTA() throws Exception {
+        request.get("/api/modeling-exercises/" + classExercise.getId() + "/statistics", HttpStatus.OK, String.class);
+        request.get("/api/modeling-exercises/" + classExercise.getId() + 1 + "/statistics", HttpStatus.NOT_FOUND, String.class);
+
+        classExercise.setDiagramType(CommunicationDiagram);
+        exerciseRepo.save(classExercise);
+        request.get("/api/modeling-exercises/" + classExercise.getId() + "/statistics", HttpStatus.NOT_FOUND, String.class);
+    }
+
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    public void testCreateModelingExercise_asInstructor() throws Exception {
+        ModelingExercise modelingExercise = modelingExerciseUtilService.createModelingExercise(classExercise.getCourse().getId());
+        request.post("/api/modeling-exercises", modelingExercise, HttpStatus.CREATED);
+
+        modelingExercise = modelingExerciseUtilService.createModelingExercise(classExercise.getCourse().getId(), 1L);
+        request.post("/api/modeling-exercises", modelingExercise, HttpStatus.BAD_REQUEST);
+
+        modelingExercise = modelingExerciseUtilService.createModelingExercise(2L);
+        request.post("/api/modeling-exercises", modelingExercise, HttpStatus.NOT_FOUND);
+    }
+
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    public void testUpdateModelingExercise_asInstructor() throws Exception {
+        ModelingExercise modelingExercise = modelingExerciseUtilService.createModelingExercise(classExercise.getCourse().getId());
+        ModelingExercise createdModelingExercise = request.putWithResponseBody("/api/modeling-exercises", modelingExercise, ModelingExercise.class, HttpStatus.CREATED);
+
+        ModelingExercise modelingExerciseWithSubmission = modelingExerciseUtilService.addExampleSubmission(createdModelingExercise);
+        ModelingExercise returnedModelingExercise = request.putWithResponseBody("/api/modeling-exercises", modelingExerciseWithSubmission, ModelingExercise.class, HttpStatus.OK);
+        assertThat(returnedModelingExercise.getExampleSubmissions().size()).isEqualTo(1);
+
+        when(courseService.findOne(classExercise.getCourse().getId())).thenReturn(null);
```

Comment:
Is this mocking behavior really necessary? I would like to avoid something like this and rather insert the data correctly into the database before executing a certain REST Call

Reply:
Yes, I added it because the `ModelingExerciseResource` checks if the course is null in the `checkModelingExercise`, even though `courseService.findOne(modelingExercise.getCourse().getId())` throws an `EntityNotFoundException` if the course is null:
https://github.com/ls1intum/Artemis/blob/0919a206cdf68a1a154ec2959d5530d320206e5d/src/main/java/de/tum/in/www1/artemis/web/rest/ModelingExerciseResource.java#L109